src/geo/locate.js

```diff
@@ -40,6 +44,68 @@ const handleQuery = async (opts) => {
   }
 }
 
+const locateCity = async ({ city, region, country }) => {
+  const query = {
+    text: `${city}, ${region} ${country}`,
+    size: 1
+  }
+
+  // check if cache has it first
+  const lruKey = JSON.stringify(query)
+  if (lru.has(lruKey)) return lru.get(lruKey)
+
+  const opts = {
+    query,
+    host: pelias.hosts.search
+  }
+  // not in cache, fetch it
+  const out = handleQuery(opts)
+  if (!out) return
+  // put it in cache for later
+  lru.set(lruKey, out)
+  return out
+}
+
+const runOverpassQuery = async (query) => {
+  const qs = `[out:json];
+    ${query}
+  out body;`
+  const { body } = await request
+    .post('http://overpass-api.de/api/interpreter')
+    .send(qs)
+    .retry(10)
+    .agent(agent)
+  return body
+}
+
+const locateWay = async ({ street, bbox }) => {
+  const query = `way
+      ["name"~"${capitalize.words(street)}"]
+      (${bbox[1]}, ${bbox[0]}, ${bbox[3]}, ${bbox[2]});`
+  return runOverpassQuery(query)
+}
+
+const lookupNodeId = async (nodeId) => {
+  const { elements } = await runOverpassQuery(`node(${nodeId});`)
+  return elements[0]
+}
+
+const intersectionSplitExp = /[/,]/
+
+const locateIntersection = async ({ intersection, city, region, country }) => {
+  const { bbox } = await locateCity({ city, region, country }) //get city's bounding box
+
+  // use bounding box in searches
+  const streets = _.map(intersection.split(intersectionSplitExp), _.trim) // split street intersections on forward slash and comma
+  const waysData = await Promise.all(_.map(streets, async (street) => {
```

Comment:
Why are you using lodash for the map?

Reply:
Performance and readability: https://www.reddit.com/r/javascript/comments/8j2hmq/performance_lodash_vs_es6_map/

Reply:
The performance between lodash and es6 map is so minuet, fractions of milliseconds over large data sets. Benchmarking nodejs is a troubling debacle since your data set size will affect the outcome of both lodash and native. If it is not orders of magnitude faster or slower don't switch to lodash.